<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ControllerWindow</class>
 <widget class="QMainWindow" name="ControllerWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>800</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1200</width>
    <height>800</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>NeuRPi Terminal - Unified Interface</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="centralwidget_layout">
    <item>
     <widget class="QTabWidget" name="main_tabs">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="control_tab">
       <attribute name="title">
        <string>Control Center</string>
       </attribute>
       <layout class="QVBoxLayout" name="control_tab_layout">
        <item>
         <widget class="QSplitter" name="main_splitter">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>

          <!-- Left Panel: Experiment Setup -->
          <widget class="QWidget" name="experiment_panel">
           <layout class="QVBoxLayout" name="experiment_panel_layout">

            <!-- Experiment Setup Group -->
            <item>
             <widget class="QGroupBox" name="experiment_setup_group">
              <property name="title">
               <string>Experiment Setup</string>
              </property>
              <layout class="QGridLayout" name="experiment_setup_layout">
               <item row="0" column="0">
                <widget class="QLabel" name="protocol_label">
                 <property name="text">
                  <string>Protocol:</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QComboBox" name="protocol_combo"/>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="experiment_label">
                 <property name="text">
                  <string>Experiment:</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QComboBox" name="experiment_combo"/>
               </item>
               <item row="2" column="0">
                <widget class="QLabel" name="configuration_label">
                 <property name="text">
                  <string>Configuration:</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="1">
                <widget class="QComboBox" name="configuration_combo"/>
               </item>
               <item row="3" column="0">
                <widget class="QLabel" name="rig_label">
                 <property name="text">
                  <string>Rig/Pilot:</string>
                 </property>
                </widget>
               </item>
               <item row="3" column="1">
                <widget class="QComboBox" name="rig_combo"/>
               </item>
              </layout>
             </widget>
            </item>

            <!-- Subject Setup Group -->
            <item>
             <widget class="QGroupBox" name="subject_setup_group">
              <property name="title">
               <string>Subject Setup</string>
              </property>
              <layout class="QGridLayout" name="subject_setup_layout">
               <item row="0" column="0">
                <widget class="QLabel" name="subject_label">
                 <property name="text">
                  <string>Subject:</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QComboBox" name="subject_combo"/>
               </item>
               <item row="0" column="2">
                <widget class="QPushButton" name="new_subject_btn">
                 <property name="text">
                  <string>New Subject</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="weight_label">
                 <property name="text">
                  <string>Current Weight:</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QDoubleSpinBox" name="weight_spinbox">
                 <property name="suffix">
                  <string> g</string>
                 </property>
                 <property name="maximum">
                  <double>1000.000000000000000</double>
                 </property>
                </widget>
               </item>
               <item row="2" column="0" colspan="2">
                <widget class="QCheckBox" name="water_restriction_check">
                 <property name="text">
                  <string>Water Restriction</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>

            <!-- Session Control Group -->
            <item>
             <widget class="QGroupBox" name="session_control_group">
              <property name="title">
               <string>Session Control</string>
              </property>
              <layout class="QHBoxLayout" name="session_control_layout">
               <item>
                <widget class="QPushButton" name="start_btn">
                 <property name="text">
                  <string>Start Experiment</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="stop_btn">
                 <property name="text">
                  <string>Stop Experiment</string>
                 </property>
                 <property name="enabled">
                  <bool>false</bool>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>

            <!-- Session Information Group -->
            <item>
             <widget class="QGroupBox" name="session_info_group">
              <property name="title">
               <string>Session Information</string>
              </property>
              <layout class="QVBoxLayout" name="session_info_layout">
               <item>
                <widget class="QPlainTextEdit" name="session_info_text">
                 <property name="maximumSize">
                  <size>
                   <width>16777215</width>
                   <height>150</height>
                  </size>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>

           </layout>
          </widget>

          <!-- Right Panel: System Status -->
          <widget class="QWidget" name="status_panel">
           <layout class="QVBoxLayout" name="status_panel_layout">

            <!-- System Status Group -->
            <item>
             <widget class="QGroupBox" name="system_status_group">
              <property name="title">
               <string>System Status</string>
              </property>
              <layout class="QVBoxLayout" name="system_status_layout">
               <item>
                <widget class="QLabel" name="status_label">
                 <property name="text">
                  <string>Terminal: Running</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>

            <!-- Connected Pilots Group -->
            <item>
             <widget class="QGroupBox" name="pilots_group">
              <property name="title">
               <string>Connected Pilots</string>
              </property>
              <layout class="QVBoxLayout" name="pilots_layout">
               <item>
                <widget class="QTableWidget" name="pilots_table">
                 <column>
                  <property name="text">
                   <string>Name</string>
                  </property>
                 </column>
                 <column>
                  <property name="text">
                   <string>IP</string>
                  </property>
                 </column>
                 <column>
                  <property name="text">
                   <string>State</string>
                  </property>
                 </column>
                 <column>
                  <property name="text">
                   <string>Last Seen</string>
                  </property>
                 </column>
                </widget>
               </item>
               <item>
                <layout class="QHBoxLayout" name="pilot_buttons_layout">
                 <item>
                  <widget class="QPushButton" name="refresh_btn">
                   <property name="text">
                    <string>Refresh</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QPushButton" name="ping_btn">
                   <property name="text">
                    <string>Ping All</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </widget>
            </item>

            <!-- Log Output Group -->
            <item>
             <widget class="QGroupBox" name="log_group">
              <property name="title">
               <string>Log Output</string>
              </property>
              <layout class="QVBoxLayout" name="log_layout">
               <item>
                <widget class="QPlainTextEdit" name="log_text"/>
               </item>
               <item>
                <layout class="QHBoxLayout" name="log_controls_layout">
                 <item>
                  <widget class="QPushButton" name="clear_log_btn">
                   <property name="text">
                    <string>Clear Log</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <spacer name="log_spacer">
                   <property name="orientation">
                    <enum>Qt::Horizontal</enum>
                   </property>
                   <property name="sizeHint" stdset="0">
                    <size>
                     <width>40</width>
                     <height>20</height>
                    </size>
                   </property>
                  </spacer>
                 </item>
                </layout>
               </item>
              </layout>
             </widget>
            </item>

           </layout>
          </widget>

         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1200</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_file">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="action_exit"/>
   </widget>
   <widget class="QMenu" name="menu_view">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="action_refresh"/>
   </widget>
   <widget class="QMenu" name="menu_help">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="action_about"/>
   </widget>
   <addaction name="menu_file"/>
   <addaction name="menu_view"/>
   <addaction name="menu_help"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="action_exit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="action_refresh">
   <property name="text">
    <string>Refresh</string>
   </property>
   <property name="shortcut">
    <string>F5</string>
   </property>
  </action>
  <action name="action_about">
   <property name="text">
    <string>About</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
